python/lsst/meas/extensions/multiprofit/rebuild_coadd_multiband.py

```diff
@@ -0,0 +1,248 @@
+from lsst.skymap import BaseSkyMap
+import pydantic
+from pydantic.dataclasses import dataclass
+
+from .fit_coadd_multiband import (
+    CatalogExposurePsfs, CatalogSourceFitterConfigData, MultiProFitSourceTask,
+)
+
+
+@dataclass(frozen=True, config=pydantic.ConfigDict(arbitrary_types_allowed=True))
+class ModelRebuilder:
+    """A rebuilder of MultiProFit models from their inputs and best-fit
+    parameter values.
+    """
+
+    catexps: list[CatalogExposurePsfs] = pydantic.Field(doc="List of catalog-exposure-psf objects")
+    catalog_multi: afwTable.SourceCatalog = pydantic.Field(
+        doc="Patch-level multiband reference catalog (deepCoadd_ref)",
+    )
+    fit_results: astropy.table.Table = pydantic.Field(doc="Multiprofit model fit results")
+    task_fit: MultiProFitSourceTask = pydantic.Field(doc="The task")
+
+    @cached_property
+    def channels(self) -> tuple[g2f.Channel]:
+        channels = tuple(g2f.Channel.get(catexp.band) for catexp in self.catexps)
+        return channels
+
+    @cached_property
+    def config_data(self) -> CatalogSourceFitterConfigData:
+        config_data = self.make_config_data()
+        return config_data
+
+    @classmethod
+    def from_quantumGraph(
+        cls,
+        butler: dafButler.Butler,
+        quantumgraph: QuantumGraph,
+        dataId: dict = None,
+    ):
+        """Make a rebuilder from a butler and quantumgraph.
+
+        Parameters
+        ----------
+        butler
+            The butler that the quantumgraph was built for.
+        quantumgraph
+            The quantum graph file from a CoaddMultibandFitTask using the
+            MultiProFitSourceTask.
+        dataId
+            The dataId for the fit, including skymap, tract and patch.
+
+        Returns
+        -------
+        rebuilder
+            A ModelRebuilder instance initialized with the necessary kwargs.
+        """
+        if dataId is None:
+            quantum = next(iter(quantumgraph.outputQuanta)).quantum
+        else:
+            quantum = None
+            for node in quantumgraph.outputQuanta:
+                if node.quantum.dataId.to_simple().dataId == dataId:
+                    quantum = node.quantum
+                    break
+            if quantum is None:
+                raise ValueError(
+                    f"{dataId=} not found in {[x.quantum.dataId for x in quantumgraph.outputQuanta]=}"
+                )
+        taskDef = next(iter(quantumgraph.iterTaskGraph()))
+        butlerQC = QuantumContext(butler, quantum)
+        config = butler.get(f"{taskDef.label}_config")
+        # I have no idea what to put for initInputs.
+        # quantum.initInputs looks wrong - the values can be lists
+        # quantumgraph.initInputRefs(taskDef) returns a list of DatasetRefs...
+        # ... but I'm not sure how to map that to connection names?
+        task: CoaddMultibandFitTask = taskDef.taskClass(config=config, initInputs={})
```

Comment:
Yeah I haven't dived deep enough into the middle ware to know what this does, but I guess you don't need it?

Reply:
It works as is, but when I originally looked in to it, I couldn't work out under which circumstances I might add a connection to the task and break this line. I guess we'll find out and I'll leave this comment here to jog my memory if that happens.